packages/idyll-components/src/cite.js

```diff
@@ -0,0 +1,112 @@
+const React = require('react');
+
+const citations = [];
+const referenceInstances = [];
+
+const citationsChangeHandler = {
+  set: function(target, property, value) {
+    referenceInstances.forEach(referenceInstance => {
+      referenceInstance.forceUpdate();
+    });
+    target[property] = value;
+    return true;
+  }
+};
+
+var citationsProxy = new Proxy(citations, citationsChangeHandler);
+
+class References extends React.Component {
+  constructor() {
+    super();
+    referenceInstances.push(this);
+  }
+  render() {
+    return (
+      <div id="references">
+        <h1>References</h1>
+        <ol>
+          {citations.map((citation, index) => (
+            <li key={index}>
+              {citation.authors}:{' '}
+              <a href={citation.url} target="_blank">
+                {citation.title}
+              </a>
+              , {citation.date}
+            </li>
+          ))}
+        </ol>
+      </div>
+    );
+  }
+}
+
+class Cite extends React.Component {
+  constructor(props) {
+    super(props);
+    const { authors, url, title, id, date } = this.props;
+    const foundCitation = citationsProxy.find(
+      citation => citation.id === id && id !== undefined
+    );
+    if (foundCitation) {
+      this.state = {
+        citationNumber: citationsProxy.indexOf(foundCitation) + 1,
+        ...foundCitation
+      };
+      if (authors || title) {
+        console.warn(
+          'Warning: Ignoring authors and title as Cite with the given id was declared before.'
+        );
+      }
+    } else if (authors && title) {
+      const newCitation = { authors, url, title, id, date };
+      citationsProxy.push(newCitation);
+      this.state = { citationNumber: citationsProxy.length, ...newCitation };
+    } else {
+      console.warn(
+        `Warning: Cite with given id not found or invalid id. Check if a Cite with the id "${id}" was declared before this line.`
+      );
+      this.state = {};
+    }
+  }
+  render() {
+    const { citationNumber, authors, title } = this.state;
+    return (
+      <a title={`${authors}: ${title}`} href="#references">
```

Comment:
Can we hyperlink to the specific citation number?

Reply:
updated